com.unity.render-pipelines.core/Editor/Lighting/ProbeVolume/ProbeVolumeSubdivide.compute

```diff
@@ -0,0 +1,261 @@
+float _ProbeVolumeCount;
+
+[numthreads(8,8,1)]
+void Clear(uint3 id : SV_DispatchThreadID)
+{
+    _Output[id] = 0;
+}
+
+[numthreads(64,1,1)]
+void ClearBuffer(uint3 id : SV_DispatchThreadID)
+{
+    _BricksToClear[id.x] = 0;
+}
+
+[numthreads(8,8,1)]
+void FillUVMap(uint3 id : SV_DispatchThreadID)
+{
+    float4 input = _Input[id];
+
+    if (input.a > 0.5)
+    {
+        _Output[id] = float4(id / _Size.x, 1);
+    }
+    else
+    {
+        _Output[id] = 0; // mark UV as invalid with w = 0
+    }
+}
+
+static float3 offset3D[27] =
+{
+    float3(-1, -1, -1),
+    float3(-1,  0, -1),
+    float3(-1,  1, -1),
+    float3( 0, -1, -1),
+    float3( 0,  0, -1),
+    float3( 0,  1, -1),
+    float3( 1, -1, -1),
+    float3( 1,  0, -1),
+    float3( 1,  1, -1),
+
+    float3(-1, -1,  0),
+    float3(-1,  0,  0),
+    float3(-1,  1,  0),
+    float3( 0, -1,  0),
+    float3( 0,  0,  0),
+    float3( 0,  1,  0),
+    float3( 1, -1,  0),
+    float3( 1,  0,  0),
+    float3( 1,  1,  0),
+
+    float3(-1, -1,  1),
+    float3(-1,  0,  1),
+    float3(-1,  1,  1),
+    float3( 0, -1,  1),
+    float3( 0,  0,  1),
+    float3( 0,  1,  1),
+    float3( 1, -1,  1),
+    float3( 1,  0,  1),
+    float3( 1,  1,  1),
+};
+
+float Distance(float3 pos)
+{
+    return length(pos);
+}
+
+[numthreads(8,8,1)]
+void JumpFlooding(uint3 id : SV_DispatchThreadID)
+{
+    float4 nearest = _Input[id];
+
+    if (nearest.w < 0.5)
+        nearest = float4(-10, -10, -10, 0);
+
+    int j = 0;
+    for (int i = 0; i < 27; i++)
+    {
+        float3 o = id + offset3D[i] * _Offset;
+
+        o %= _Size.x;
+        o.x += (o.x < 0) ? _Size.x : 0;
+        o.y += (o.y < 0) ? _Size.x : 0;
+        o.z += (o.z < 0) ? _Size.x : 0;
+        float4 n1 = _Output[o];
+
+        // Discard invalid samples
+        if (n1.w < 0.5)
+            continue;
+
+        float3 uv = id / _Size.x;
+        if (Distance(uv - n1.xyz) < Distance(uv - nearest.xyz))
+            nearest = n1;
+
+    }
+    _Output[id] = nearest;
+}
+
+[numthreads(8, 8, 1)]
+void FinalPass(uint3 id : SV_DispatchThreadID)
+{
+    float3 defaultUV = id / _Size.x;
+    float3 nearestUV = _Input[id].xyz;
+
+    float dist = Distance(nearestUV - defaultUV);
+
+    _Output[id] = float4(dist.xxx, 1);
+}
+
+float2 ProjectOBB(GPUProbeVolumeOBB obb, float3 axis)
+{
+    float pMin = dot(axis, obb.corner);
+    float pMax = pMin;
+
+    for (int x = 0; x < 2; x++)
+    {
+        for (int y = 0; y < 2; y++)
+        {
+            for (int z = 0; z < 2; z++)
+            {
+                float3 vert = obb.corner + obb.X * x + obb.Y * y + obb.Z * z;
+
+                float proj = dot(axis, vert);
+
+                if (proj < pMin)
+                    pMin = proj;
+                else if (proj > pMax)
+                    pMax = proj;
+            }
+        }
+    }
+
+    return float2(pMin, pMax);
+}
+
+bool PointInsideOBB(GPUProbeVolumeOBB obbA, GPUProbeVolumeOBB obbB)
+{
+    float3 axes[] =
```

Comment:
It is fine because this is a bake time shader, but worth pointing that generally this pattern (declaring local arrays in hlsl) leads to awful code generation